apps/mobile-wallet/src/features/ecosystem/DAppWebViewScreen.tsx

```diff
@@ -0,0 +1,74 @@
+import { NativeStackScreenProps } from '@react-navigation/native-stack'
+import * as Clipboard from 'expo-clipboard'
+import { useEffect } from 'react'
+import { useTranslation } from 'react-i18next'
+import WebView from 'react-native-webview'
+import styled from 'styled-components/native'
+
+import Screen, { ScreenProps } from '~/components/layout/Screen'
+import { useWalletConnectContext } from '~/contexts/walletConnect/WalletConnectContext'
+import AddToFavoritesButton from '~/features/ecosystem/AddToFavoritesButton'
+import { activateAppLoading, deactivateAppLoading } from '~/features/loader/loaderActions'
+import { useAppDispatch } from '~/hooks/redux'
+import RootStackParamList from '~/navigation/rootStackRoutes'
+import { HEADER_OFFSET_TOP } from '~/style/globalStyle'
+
+interface DAppWebViewScreenProps extends NativeStackScreenProps<RootStackParamList, 'DAppWebViewScreen'>, ScreenProps {}
+
+const DAppWebViewScreen = ({ navigation, route, ...props }: DAppWebViewScreenProps) => {
+  const { dAppUrl, dAppName } = route.params
+
+  useDetectWCUrlInClipboardAndPair()
+
+  if (!dAppUrl) return null
+
+  return (
+    <Screen
+      titleAlwaysVisible
+      headerOptions={{
+        type: 'stack',
+        headerTitle: dAppUrl,
+        headerRight: () => <AddToFavoritesButton dAppName={dAppName} />
+      }}
+      {...props}
+    >
+      <WebViewStyled source={{ uri: dAppUrl }} allowsBackForwardNavigationGestures pullToRefreshEnabled />
+    </Screen>
+  )
+}
+
+export default DAppWebViewScreen
+
+const useDetectWCUrlInClipboardAndPair = () => {
+  const dispatch = useAppDispatch()
+  const { t } = useTranslation()
+  const { pairWithDapp } = useWalletConnectContext()
+
+  useEffect(() => {
+    const checkClipboard = async () => {
+      const content = await Clipboard.getStringAsync()
+
+      if (content.startsWith('wc:')) {
+        Clipboard.setStringAsync('')
+
+        // TODO: Check if the feature is enabled and warn the user? Or do we just drop the experimental warning finally?
```

Comment:
That's a good question.
If we're looking into JS injection, I'd rather keep this experimental, hoping JS injection could just replace WC by default.

Reply:
If we keep it experimental, then users who have the feature off (new installs) will need to exit the dApp webbview and the ecosystem screen to go to settings, enable it, and come back here. I find it a bit useless at this point and I'd prefer removing the whole "experimental" thing and enable WC by default.